lib/membrane_live_web/plugs/auth.ex

```diff
@@ -6,28 +6,25 @@ defmodule MembraneLiveWeb.Plugs.Auth do
   import Plug.Conn
   alias MembraneLive.Tokens
   alias MembraneLiveWeb.FallbackController
+  alias MembraneLiveWeb.Helpers.TokenErrorInfo
 
+  @auth_header_key "authorization"
+
+  @spec init(any) :: any
   def init(default), do: default
 
   def call(conn, _default) do
-    with {"authorization", "Bearer " <> token} <- find_bearer(conn.req_headers),
-         {:ok, %{"user_id" => user_id}} <- Tokens.custom_decode(token) do
+    with {@auth_header_key, "Bearer " <> token} <-
+           find_header_value(conn.req_headers, @auth_header_key),
+         {:ok, %{"user_id" => user_id}} <- Tokens.auth_decode(token) do
       assign(conn, :user_id, user_id)
     else
       err ->
-        FallbackController.call(conn, get_error_info(err))
+        FallbackController.call(conn, TokenErrorInfo.get_error_info(err))
         |> halt()
     end
   end
 
-  defp find_bearer(headers),
-    do: Enum.find(headers, fn elem -> match?({"authorization", _value}, elem) end)
-
-  defp get_error_info(nil), do: %{error: :bad_request, message: "Lack of authentication data"}
-
-  defp get_error_info({:error, :signature_error}),
-    do: %{error: :unauthorized, message: "Token has an invalid signature"}
-
-  defp get_error_info({:error, _error_reason}),
-    do: %{error: :unauthorized, message: "Unknown token validation error"}
+  defp find_header_value(headers, key),
+    do: Enum.find(headers, fn elem -> match?({^key, _value}, elem) end)
```

Comment:
```suggestion
  defp find_header_value(headers, key),
    do: Enum.find(headers, fn 
    {^key, _value} -> true 
    _elem -> false 
  end)
```

Reply:
I'll probably change it even more. I don't like the fact this function returns `nil` if the header is absent. Give me 10 min 🕚